service-block/src/main/java/blocks/service/ServiceProperties.java

```diff
@@ -0,0 +1,71 @@
+package blocks.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.IntNode;
+import com.fasterxml.jackson.databind.node.LongNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class ServiceProperties {
+
+    public final Map<String, JsonNode> configuredProperties;
+
+    private ServiceProperties(final Map<String, JsonNode> systemProperties) {
+        this.configuredProperties = Collections.unmodifiableMap(systemProperties);
+    }
+
+    public static class Builder {
+
+        private static final Map<Class<?>, Function<Object, JsonNode>> objectToNode = new HashMap<>() {{
+            put(java.lang.String.class, o -> TextNode.valueOf((String) o));
+            put(java.lang.Long.class, o -> LongNode.valueOf((long) o));
+            put(java.lang.Integer.class, o -> IntNode.valueOf((int) o));
+            put(java.time.LocalDateTime.class, o -> TextNode.valueOf(((LocalDateTime) o).toString()));
+            put(java.time.ZonedDateTime.class, o -> TextNode.valueOf(((ZonedDateTime) o).toString()));
```

Comment:
This way of formatting (toString()) may not necessarily be in line with what someone would want this property to show as in info endpoint and websocket stream... Would be good to think about allowing people to submit formatter if needed or also add conversion that when sees subclass of JsonNode in the value will convert it to itself (essentially allowing passing in preformatted JsonNode.